includes/class-duplicate-payment-prevention-service.php

```diff
@@ -150,6 +150,14 @@ public function check_against_session_processing_order( WC_Order $current_order
 			return;
 		}
 
+		if ( ! $current_order->has_status( wc_get_is_pending_statuses() ) ) {
+			return;
+		}
+
+		if ( $session_order->get_id() === $current_order->get_id() ) {
+			return;
+		}
+
```

Comment:
May add this check too. It does not help check orders from different guests but it's still good enough to check (1) among registered customers, and (2) registered customers vs any guest. 

```suggestion

		if ( $session_order->get_customer_id() !== $current_order->get_customer_id() ) {
			return;
		}
```